feat(docs): add coc-rust-analyzer (#1755)
rust-analyzer/rust-analyzer: docs/user/README.md

  - Note that if you use a plugin manager other than `vim-plug`, you may need to manually
    checkout the `release` branch wherever your plugin manager cloned it. Otherwise you will
    get errors about a missing javascript file.

> * Run `:CocInstall coc-rust-analyzer` to install `coc-rust-analyzer`, this extension already implemented _almost_ all features supported in VSCode extension:

Reviewer: ```suggestion
* Run `:CocInstall coc-rust-analyzer` to install [coc-rust-analyzer], this extension implements _most_ of the features supported in the VSCode extension:
```

  - same configurations as VSCode extension, `rust-analyzer.raLspServerPath`, `rust-analyzer.enableCargoWatchOnStartup` etc.
  - same commands too, `rust-analyzer.analyzerStatus`, `rust-analyzer.startCargoWatch` etc.

>   - highlighting and inlay_hints is not works by now

Reviewer: ```suggestion
  - highlighting and inlay_hints is not yet implemented
```

[coc.nvim]: https://github.com/neoclide/coc.nvim
[coc-vim-conf]: https://github.com/neoclide/coc.nvim/#example-vim-configuration

> 

Reviewer: ```suggestion
[coc-rust-analyzer]: https://github.com/fannheyward/coc-rust-analyzer
```
